telink: tl321x: update for SOC version 2

haiwentelink opened the pull request:
- update to V4.0.4.3
- add pwm module in compilation.
- add USB CDC sample in ci build.
- use adc_get_code() to fix reading errors.
- update to 60m mspi to save power.
- clean code for comment code.
- update hal_telink in west.yml.

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/8e116bb74c344c043500425793d76c4d3ae21565 | https://github.com/telink-semi/hal_telink/commit/38c66795ef4a79453330a1aed85d1217843a8d05 (develop) | [telink-semi/hal_telink@8e116bb7..38c66795](https://github.com/telink-semi/hal_telink/compare/8e116bb74c344c043500425793d76c4d3ae21565..38c66795ef4a79453330a1aed85d1217843a8d05) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

s07641069 reviewed (changes requested):
you modified hash of hal_intel instead of hal_telink

s07641069 reviewed (changes requested):
why drivers/adc/adc_tlx.c is without modifications as you modified in https://github.com/telink-semi/zephyr/pull/456/files ?

haiwentelink commented:
@s07641069  for buteo ram is limit , and in modules/lib/openthread, buf is very large , and i checked with matter setting , so change it to the same as the setting of the matter in .github/workflows/telink-tl321x-build.yaml.

s07641069 commented:
@haiwentelink 
Please fix last compliance checks warning (see https://github.com/telink-semi/zephyr/pull/486/files)
and after merge PR

haiwentelink committed:
telink: tl321x: update for SOC version 2

- update to V4.0.4.3
- add pwm module in compilation.
- add USB CDC sample in ci build.
- use adc_get_code() to fix reading errors.
- update to 60m mspi to save power.
- clean code for comment code.
- update hal_telink in west.yml.

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>
